### ISSUE TITLE
odd behavior with go1.9...

### ISSUE THREAD

**Issue by kevinburke** [dominikh/go-tools#165]
Spotted in Travis. Tests alone run fine so I think this is an issue with staticcheck... see for example https://travis-ci.org/kevinburke/h256only/jobs/268255688. It can't find stdlib packages anywhere.

```
/home/travis/gopath/bin/staticcheck ./...
/home/travis/gopath/src/github.com/kevinburke/h256only/claims.go:4:2: could not import crypto/subtle (cannot find package "crypto/subtle" in any of:
	/usr/local/go/src/crypto/subtle (from $GOROOT)
	/home/travis/gopath/src/crypto/subtle (from $GOPATH))
/home/travis/gopath/src/github.com/kevinburke/h256only/errors.go:4:2: could not import errors (cannot find package "errors" in any of:
	/usr/local/go/src/errors (from $GOROOT)
	/home/travis/gopath/src/errors (from $GOPATH))
/home/travis/gopath/src/github.com/kevinburke/h256only/hmac.go:4:2: could not import crypto/hmac (cannot find package "crypto/hmac" in any of:
	/usr/local/go/src/crypto/hmac (from $GOROOT)
	/home/travis/gopath/src/crypto/hmac (from $GOPATH))
/home/travis/gopath/src/github.com/kevinburke/h256only/hmac.go:5:2: could not import crypto/sha256 (cannot find package "crypto/sha256" in any of:
	/usr/local/go/src/crypto/sha256 (from $GOROOT)
	/home/travis/gopath/src/crypto/sha256 (from $GOPATH))
```

**Comment by kevinburke** [comment]
here's a build that only runs `go install` and `go test -race` that passes so the problem may be unique to Go 1.9 and staticcheck https://travis-ci.org/kevinburke/h256only/builds/268256248

**Comment by mvdan** [comment]
Perhaps there's something weird with how Travis sets up 1.9? Or perhaps the release tarball is somehow weird? I run these tools on tip fairly regularly, and I've never encountered this issue.

**Comment by HeavyHorst** [comment]
I have the same issue on travis and go1.9.
Works fine locally with go 1.9 and also on travis with tip.

**Comment by dominikh** [comment]
What are the outputs of `go env`, `go list std` and `tree $(go env GOROOT)/src`?

**Comment by bradleyfalzon** [comment]
https://travis-ci.org/bradleyfalzon/travis/builds/268329094

**Comment by dominikh** [comment]
The combination of

> $ go env
> GOROOT="/home/travis/.gimme/versions/go1.9.linux.amd64"

and

> $ staticcheck ./...
> /home/travis/gopath/src/github.com/bradleyfalzon/travis/main.go:3:8: could not import regexp (cannot find package "regexp" in any of:
> 	/usr/local/go/src/regexp (from $GOROOT)
> 	/home/travis/gopath/src/regexp (from $GOPATH))
> couldn't load packages due to errors: github.com/bradleyfalzon/travis

suggests to me that the Go installation in Travis is broken. The value of runtime.GOROOT is `/usr/local/go`, but the Go installation really lives somewhere in `/home`. My best guess is that gimme rewrites the embedded value of GOROOT in one place, but forgets to update the sources of the runtime package. Staticcheck, in turn, is built with the wrong GOROOT and can't find the standard library sources.

**Comment by dominikh** [comment]
@bradleyfalzon one more thing to check, just to be sure: `echo $GOROOT` and `env | grep GOROOT`

**Comment by dominikh** [comment]
And finally: `grep DefaultGoroot $(go env GOROOT)/src/runtime/internal/sys/zversion.go`

**Comment by bradleyfalzon** [comment]
https://travis-ci.org/bradleyfalzon/travis/builds/268344370

```
The command "tree $(go env GOROOT)/src" exited with 0.
$ echo $GOROOT
The command "echo $GOROOT" exited with 0.
$ env | grep GOROOT
The command "env | grep GOROOT" exited with 1.
$ grep DefaultGoroot $(go env GOROOT)/src/runtime/internal/sys/zversion.go
const DefaultGoroot = `/usr/local/go`
```

**Comment by dominikh** [comment]
As I suspected, the value of DefaultGoroot in the runtime package is incorrect, and a correct GOROOT environment variable isn't being exported to the environment, either. Gimme is giving you a broken Go installation.

**Comment by dominikh** [comment]
https://github.com/travis-ci/gimme/commit/b76e1a0d6ad6af6e559621f5960a7eec3b03fe63 is likely the fix for this issue. However, Travis seems to be using gimme 1.0, whereas the fix exists in 1.2.

Either way, this is unfortunately Travis's fault. You'll want to ask them to fix it. Closing this issue as "not our fault"

**Comment by dominikh** [comment]
Upstream issue, including workaround: https://github.com/travis-ci/travis-ci/issues/8229
